tests/baselines/reference/genericRestParameters3.errors.txt

```diff
@@ -6,20 +6,20 @@ tests/cases/conformance/types/rest/genericRestParameters3.ts(18,1): error TS2345
 tests/cases/conformance/types/rest/genericRestParameters3.ts(22,1): error TS2322: Type '(x: string, ...args: [string] | [number, boolean]) => void' is not assignable to type '(...args: [string, string] | [string, number, boolean]) => void'.
 tests/cases/conformance/types/rest/genericRestParameters3.ts(23,1): error TS2322: Type '(x: string, y: string) => void' is not assignable to type '(x: string, ...args: [string] | [number, boolean]) => void'.
   Types of parameters 'y' and 'args' are incompatible.
-    Type '[string] | [number, boolean]' is not assignable to type '[string]'.
-      Type '[number, boolean]' is not assignable to type '[string]'.
+    Type '[string] | [number, boolean]' is not assignable to type '[y: string]'.
+      Type '[number, boolean]' is not assignable to type '[y: string]'.
         Types of property '0' are incompatible.
           Type 'number' is not assignable to type 'string'.
 tests/cases/conformance/types/rest/genericRestParameters3.ts(24,1): error TS2322: Type '(x: string, y: number, z: boolean) => void' is not assignable to type '(x: string, ...args: [string] | [number, boolean]) => void'.
   Types of parameters 'y' and 'args' are incompatible.
-    Type '[string] | [number, boolean]' is not assignable to type '[number, boolean]'.
-      Property '1' is missing in type '[string]' but required in type '[number, boolean]'.
+    Type '[string] | [number, boolean]' is not assignable to type '[y: number, z: boolean]'.
+      Property '1' is missing in type '[string]' but required in type '[y: number, z: boolean]'.

```

Comment:
Hmm, the _actual_ property name is `'1'`, but it sure looks like it's name is `z`. Wonder if it is better to omit the names in some cases?

Reply:
Eh... I think it'd be kinda odd to sometimes display the same type with labels and sometimes without. Plus, this error is somewhat rare; I believe we usually try to report tuple length errors as an arity mismatch (ie, on length), but fail to do so here because of the union.

Reply:
How about 
```
Type '[string] | [number, boolean]' is not assignable to type '[y: number, z: boolean]'.
      Property '1' (labeled as 'z') is missing in type '[string]' but required in type '[y: number, z: boolean]'.
```

Sorry for the late question, I was super excited for this feature as I want to use it now (and get the benefit in future as Angular updates it's TS dependency)